plugin/documents.py

```diff
@@ -383,6 +383,19 @@ def on_post_save_async(self) -> None:
             # The URI scheme has changed. This means we need to re-determine whether any language servers should
             # be attached to the view.
             sublime.set_timeout(self._reset)
+        # Hide empty diagnostics panels on save if the show_diagnostics_panel_on_save option is enabled.
+        if userprefs().show_diagnostics_panel_on_save > 0:
+            self._hide_empty_diagnostics_panels()
+
+    def _hide_empty_diagnostics_panels(self) -> None:
+        severity_threshold = userprefs().show_diagnostics_severity_level
+        hide_panel = True
+        for sb, diagnostics in self.diagnostics_async():
+            for diagnostic, _ in diagnostics:
+                if diagnostic_severity(diagnostic) <= severity_threshold:
+                    hide_panel = False
```

Comment:
Must only hide the panel if there is no diagnostics that match the requested severity. Not like now where it decides to hide on first diagnostic that it sees that doesn't match severity.

Also the logic should also close the panel if there are no diagnostics at all.

And lastly, this is comparing with the wrong setting. Should compare severity to `show_diagnostics_panel_on_save`, not `show_diagnostics_severity_level`.

Reply:
Sorry, the logic was correct already but the preference name was not.